resources/assets/scripts/components/accordion.vue

```diff
@@ -0,0 +1,86 @@
+<template>
+  <div class="accordion">
+    <a :href="href" @click.prevent="toggle" :id="`${localId}-title`" :aria-controls="`${localId}-content`" :aria-expanded="active" :role="role">
+      <slot name="title"></slot>
+    </a>
+    <div v-show="active" :id="`${localId}-content`" :aria-labelledby="`${localId}-title`" :aria-hidden="!active" :role="role === 'tab' ? 'tabpanel' : null">
+      <slot></slot>
+    </div>
+  </div>
+</template>
+
+<style>
+</style>
+
+<script>
+import { MixinId } from 'vue-a11y-utils';
+
+export default {
+  mixins: [MixinId],
+  props: {
+    deepLink: {
+      type: Boolean,
+      default: false,
+    },
+    id: [String, Number],
+    role: String,
+  },
+  data() {
+    return {
+      active: false,
+    };
+  },
+  computed: {
+    href() {
+      return this.deepLink ? `#${this.localId}-content` : '#';
+    },
+    isTablist() {
+      return this.$parent.$el;
+    }
+  },
+  methods: {
+    toggle() {
+      this.active ? this.close() : this.open();
+    },
+
+    close() {
+      this.active = false;
+      this.$parent.$emit('closed', this);
+    },
+
+    open() {
+      this.active = true;
+      this.$parent.$emit('opened', this);
+
+      if (this.href !== '#') {
+        this.updateHash();
+      }
+    },
+
+    updateHash() {
+      window.history.replaceState({}, '', this.href);
```

Comment:
By using `replaceState` we avoid scrolling the browser to the anchor link.

Reply:
i like this